src/Features/CSharp/Portable/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessAnalyzer.cs

```diff
@@ -0,0 +1,171 @@
+            DiagnosticSeverity.Hidden,
+            isEnabledByDefault: true,
+            customTags: DiagnosticCustomTags.Unnecessary);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(descriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.IfStatement);
+        }
+
+        private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
+        {
+            // look for the form "if (a != null)" or "if (null != a)"
+            var ifStatement = (IfStatementSyntax)syntaxContext.Node;
+            if (!ifStatement.Condition.IsKind(SyntaxKind.NotEqualsExpression))
+            {
+                return;
+            }
+
+            if (ifStatement.Else != null)
+            {
+                return;
+            }
+
+            var binaryExpression = (BinaryExpressionSyntax)ifStatement.Condition;
+            if (!IsNotEqualsTest(binaryExpression.Left, binaryExpression.Right) &&
+                !IsNotEqualsTest(binaryExpression.Right, binaryExpression.Left))
+            {
+                return;
+            }
+
+            // Check for both:  "if (...) { a(); }" and "if (...) a();"
+            var statement = ifStatement.Statement;
+            if (statement.IsKind(SyntaxKind.Block))
+            {
+                var block = (BlockSyntax)statement;
+                if (block.Statements.Count != 1)
+                {
+                    return;
+                }
+
+                statement = block.Statements[0];
+            }
+
+            if (!statement.Parent.IsKind(SyntaxKind.Block))
+            {
+                return;
+            }
+
+            if (!statement.IsKind(SyntaxKind.ExpressionStatement))
+            {
+                return;
+            }
+
+            var expressionStatement = (ExpressionStatementSyntax)statement;
+
+            // Check that it's of the form: "if (a != null) { a(); }}
+            var invocationExpression = ((ExpressionStatementSyntax)statement).Expression;
+            if (!invocationExpression.IsKind(SyntaxKind.InvocationExpression))
+            {
+                return;
+            }
+
+            var expression = ((InvocationExpressionSyntax)invocationExpression).Expression;
+            if (!expression.IsKind(SyntaxKind.IdentifierName))
+            {
+                return;
+            }
+
+            var conditionName = binaryExpression.Left is IdentifierNameSyntax
+                ? (IdentifierNameSyntax)binaryExpression.Left
+                : (IdentifierNameSyntax)binaryExpression.Right;
+
+            var invocationName = (IdentifierNameSyntax)expression;
+            if (!Equals(conditionName.Identifier.ValueText, invocationName.Identifier.ValueText))
+            {
+                return;
+            }
+
+            // Now make sure the previous statement is "var a = ..."
```

Comment:
Potentially.  The issue there is that this could cause a change of behavior, esp if these calls change state.  But i imagine that's so rare as to be unimportant.